### ISSUE TITLE
Support fargate/SSM Sessions

### ISSUE THREAD

**Issue by jjshoe** [aws/amazon-ssm-agent#138]
My use case is to allow me to easily SSH to instances inside of my private VPC using SSM Sessions. The biggest hurdle to getting this running is a different metadata url, as well as certian metadata items not being available in fargate, like InstanceId.

I've attached a dockerfile to make it easy to test.

**Comment by jjshoe** [comment]
[Dockerfile.txt](https://github.com/aws/amazon-ssm-agent/files/2573990/Dockerfile.txt)

**Comment by YujiaozhAws** [comment]
Hi @jjshoe,
 
Thank you for the feedback. We'll look into this issue.

**Comment by bnusunny** [comment]
Another use case is to have a lite version of cloud shell.

**Comment by hiimtu** [comment]
Hi, if I am understanding your use case, you would like to use SSM session manager to launch a session inside of ECS/Fargate container?

If so, our team is working on this feature currently and supporting containers in general.

**Comment by AlessandroEmm** [comment]
Why has this been closed?

**Comment by rdkls** [comment]
Yep curious as well - @hiimtu why is this closed?
Especially with [ssm agent now supporting port forwarding](https://aws.amazon.com/blogs/aws/new-port-forwarding-using-aws-system-manager-sessions-manager/) could be useful for total serverless ephemeral port forwarding

**Comment by hiimtu** [comment]
Didn't mean to close. Apologies!

**Comment by btorretta** [comment]
My use case for ssm access on fargate is to run chaos experiments: https://github.com/adhorn/chaos-ssm-documents

**Comment by andrewodri** [comment]
Here is a bit of a janky/expensive workaround I just started using... It _seems_ to be doing the trick though: https://gist.github.com/andrewodri/1403c0e28503051e26b24428f1ae49b9

Basically you're just registering your Fargate instance as if it were an on-prem server in a hybrid environment.

**Comment by holyjak** [comment]
Nice!

You pay for each registered instance, right? So implementing removal of
unused Registrations will be quite important, since you get a new one upon
each deployment or scale down.

**Comment by andrewodri** [comment]
@holyjak Yup, USD$5 a pop, per month ☹️ Hopefully our friends at @aws can bake in native Fargate support sometime soon...

Chances are, until then, I'll be using this pretty heavily. I think once the basic functionality and error handling are sorted, I'll bake the SNS/Lambda cleanup and service registration into a CDK script and then dump this into a proper GitHub repo.

**Comment by greenreign** [comment]
Is this on the roadmap?  This would be a huge factor in my advocacy to switch off of kubernetes and onto a pure aws stack.
